aspects/aspects.go

```diff
@@ -507,7 +519,14 @@ func get(subKeys []string, index int, node map[string]json.RawMessage, result in
 // in which case, a nested JSON object is created for each sub-key found after a dot.
 func (s JSONDataBag) Set(path string, value interface{}) error {
 	subKeys := strings.Split(path, ".")
-	_, err := set(subKeys, 0, s, value)
+
+	var err error
+	if value == nil {
+		_, err = unset(subKeys, 0, s)
```

Comment:
Is there a reason for combining the behaviour of the two? Also I feel like this should be documented in the function docs.

Reply:
Yes, it's according to the specification which defines the same methods as the current configuration (`snap get/set`). I'll add a note about nil values to the function doc